mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusLifeCycleTest.java

```diff
@@ -0,0 +1,262 @@
+
+    @BeforeAll
+    static void setUp() {
+        rabbitMQExtension.beforeAll(null);
+    }
+
+    @AfterAll
+    static void tearDown() {
+        rabbitMQExtension.afterAll(null);
+    }
+
+    @Nested
+    class SingleEventBus {
+
+        private RabbitMQEventBus eventBus;
+
+        @BeforeEach
+        void setUp() {
+            eventBus = testExtension.newEventBus();
+        }
+
+        @AfterEach
+        void tearDown() {
+            eventBus.stop();
+        }
+
+        @Test
+        void startShouldCreateEventExchange() {
+            eventBus.start();
+            assertThat(testExtension.rabbitMQManagement.listExchanges())
+                .filteredOn(exchange -> exchange.getName().equals(MAILBOX_EVENT_EXCHANGE_NAME))
+                .hasOnlyOneElementSatisfying(exchange -> {
+                    assertThat(exchange.isDurable()).isTrue();
+                    assertThat(exchange.getType()).isEqualTo(DIRECT_EXCHANGE);
+                });
+        }
+
+        @Disabled("queue is declared with durable = true but management API always return durable = false. Suspecting a bug" +
+            " from rabbitmq driver")
+        @Test
+        void startShouldCreateKeyRegistrationWorkQueue() {
+            eventBus.start();
+            String registrationKeyWorkQueueName = eventBus.getKeyRegistrationHandler().getRegistrationQueue().asString();
+            assertThat(testExtension.rabbitMQManagement.listQueues())
+                .filteredOn(queue -> queue.getName().equals(registrationKeyWorkQueueName))
+                .hasOnlyOneElementSatisfying(queue -> {
+                    assertThat(queue.isDurable()).isTrue();
+                    assertThat(queue.isExclusive()).isTrue();
+                    assertThat(queue.isAutoDelete()).isTrue();
+                });
+        }
+
+        @Test
+        void stopShouldNotDeleteEventBusExchange() {
+            eventBus.start();
+            eventBus.stop();
+
+            assertThat(testExtension.rabbitMQManagement.listExchanges())
+                .anySatisfy(exchange -> exchange.getName().equals(MAILBOX_EVENT_EXCHANGE_NAME));
+        }
+
+        @Test
+        void stopShouldNotDeleteGroupRegistrationWorkQueue() {
+            eventBus.start();
+            GroupRegistration groupRegistration = (GroupRegistration) eventBus.register(mock(MailboxListener.class), new GroupA());
+            String workQueueName = groupRegistration.getQueueName().asString();
+            eventBus.stop();
+
+            assertThat(testExtension.rabbitMQManagement.listQueues())
+                .anySatisfy(queue -> queue.getName().equals(workQueueName));
+        }
+
+        @Test
+        void stopShouldDeleteKeyRegistrationWorkQueue() {
+            eventBus.start();
+            eventBus.stop();
+
+            String registrationKeyWorkQueueName = eventBus.getKeyRegistrationHandler().getRegistrationQueue().asString();
```

Comment:
Spying the internal state of the EventBus here is bad and should be avoided...

Let me propose : 

```
// startShouldCreateATemporaryQueue
eventBus.start();

assertThat(testExtension.rabbitMQManagement.listQueues())
   .filteredOn(//Filter out none interesting queues like group queues)
   .hasSize(1);
```

\+ 

```
//stopShouldDeleteTemporaryQueue

eventBus.start();
eventBus.stop();

assertThat(testExtension.rabbitMQManagement.listQueues())
   .filteredOn(//Filter out none interesting queues like group queues)
   .isEmpty();
```